ci/do_ci.sh

```diff
@@ -11,9 +11,10 @@ if [[ "$1" == "fix_format" || "$1" == "check_format" || "$1" == "check_repositor
   build_setup_args="-nofetch"
 fi
 
+pushd $PWD
```

Comment:
`pushd $PWD` and `popd` below does nothing?

Reply:
they make sure the execution resumes back to the directory that we were before running the two setup scripts. I believe that's similar to what `cd $ENVOY_SRCDIR` did.

Reply:
This does nothing, $PWD is always point to current directory so `pushd $PWD` is no-op and `popd` is no op in that case.

Reply:
I'm fine dropping it but my test case showed it is working:

```
$ pushd $PWD
/tmp /tmp
$ mkdir testdir
$ cd testdir/
$ popd
/tmp
$ pwd
/tmp
```

Reply:
Well that testifies they do nothing because you didn't change directory and didn't stack anything at the end.

Reply:
Hmm? If someone later adds something like `mkdir -p some/dir && cd some/dir` to `ci/build_setup.sh` (for example), the execution continues from `some/dir`. `popd` ensures the execution gets back to the original dir because otherwise many subsequent steps in `do_ci.sh` would fail. I believe that was the purpose of `cd $ENVOY_SRCDIR` too, no? That cannot be used anymore because the idea is to separate workspace dir from Envoy source dir.

Reply:
If you want to ensure that, do it clearly like:
```
SRCDIR="${PWD}"
. "$(dirname "$0")"/setup_cache.sh
. "$(dirname "$0")"/build_setup.sh $build_setup_args
cd "${SRCDIR}"
```

Reply:
> otherwise if someone later adds pushd into build_setup.sh you cannot recover.

and forgets to `popd`, yes true. I'll change it to what you suggested.